dali/ft/daftformat.cpp

```diff
@@ -58,6 +60,29 @@ CPartitioner::CPartitioner()
     partitioning = false;
 }
 
+void CPartitioner::setAbort(IAbortRequestCallback * _abort)
+{
+    abortChecker = _abort;
+}
+
+bool CPartitioner::isAborting()
+{
+    if (aborting)
+        return true;
+
+    unsigned nowTick = msTick();
+    if (abortChecker && (nowTick - lastAbortCheckTick >= abortCheckFrequency))
+    {
+        if (abortChecker->abortRequested())
+        {
+            LOG(MCdebugInfo, unknownJob, "Abort requested via callback");
+            aborting = true;
+        }
+        lastAbortCheckTick = nowTick;
+    }
+    return aborting;
+}
```

Comment:
This is duplicating the functionality (from filecopy.cpp).
It would be better to call back to the filesprayer's implementation.

You can implement a IAbortRequestCallback in the filespayer that you pass to setAbort, instead of the original one.

Reply:
Reworked